crates/wascc-provider/src/states/starting.rs

```diff
@@ -138,25 +143,62 @@ state!(
 
         let mut container_handles = HashMap::new();
         for container in pod.containers() {
-            let port_assigned =
-                assign_container_port(Arc::clone(&pod_state.shared.port_map), &pod, &container)
-                    .await?;
+            let port_assigned = match assign_container_port(
+                Arc::clone(&pod_state.shared.port_map),
+                &pod,
+                &container,
+            )
+            .await
+            {
+                Ok(port) => port,
+                Err(e) => {
+                    error!("{:?}", e);
+                    let error_state = Error {
+                        message: e.to_string(),
+                    };
+                    return Ok(Transition::Error(Box::new(error_state)));
+                }
+            };
             debug!(
                 "New port assigned to {} is: {}",
                 container.name(),
                 port_assigned
             );
 
-            let container_handle = start_container(pod_state, &container, &pod, port_assigned)
-                .await
-                .unwrap();
+            let container_handle =
+                match start_container(pod_state, &container, &pod, port_assigned).await {
+                    Ok(handle) => handle,
+                    Err(e) => {
+                        // TODO: identify if any of the start_container errors are fatal
```

Comment:
Looks like two of the errors are checking things that _shouldnt_ every be missing. I wonder if there is a better way to engineer `PodState` to make those impossible. Spawn appears to only panic from outside a runtime, so I think that is something that would be a critical error.

Reply:
Yes, again I dithered about this, and ended up thinking that if we transitioned to Error then at least we would have the pods in a known state.  The failed pod will then go back to Registered, though, and I don't know if that restart has any chance of success - if not, then we should indeed just return these errors back to the run loop and let it mark the pod as Failed and go home,